### PR TITLE
[tuner] Add CI for all tuning PR

### PR DESCRIPTION
**Pull request by RattataKing**
- Add `pre-commit.yml` and `pytest.yml`
- Remove `test_walk_mlir_op()` in `test_tune.py`, which has dependencies on iree compiler, and its test data (template input mlir) is hard to maintain
- Remove `test-data` contents

### CODE REVIEW

File: .github/workflows/pre-commit.yml
Comment (ScottTodd) [inline .github/workflows/pre-commit.yml:29]: Did you consider using https://github.com/pre-commit/action, instead of this explicit coding? The options used are a bit different, see https://github.com/pre-commit/action/blob/main/action.yml:
* `pre-commit run --all-files`
* `pre-commit run --show-diff-on-failure --color=always ${{ inputs.extra_args }}`

I think you should still be able to use the `working-directory` with that action

File: .github/workflows/pre-commit.yml
Comment (ScottTodd) [inline .github/workflows/pre-commit.yml:3]: May want to limit this to only pull requests affecting the `tuning` subdirectory

```yml
on:
  pull_request:
    paths:
      - "tuning/*"
```

File: .github/workflows/pre-commit.yml
Comment (ScottTodd) [inline .github/workflows/pre-commit.yml:1]: There are some diffs from running pre-commit that should be resolved either prior to this PR or on this PR itself, so future PRs don't have unrelated lint errors.

https://github.com/nod-ai/sdxl-scripts/actions/runs/10206491775/job/28239507156

(also use `--show-diff-on-failure` like https://github.com/pre-commit/action does so the workflow logs are more helpful)

File: .github/workflows/pytest.yml
Comment (kuhar) [inline .github/workflows/pytest.yml:47]: A few too many newlines here
```suggestion
```

File: .github/workflows/pytest.yml
Comment (kuhar) [inline .github/workflows/pytest.yml:43]: Can we make pytest discover all tests on its own?
```suggestion
          python -m pytest
```

File: .github/workflows/pre-commit.yml
Comment (kuhar) [inline .github/workflows/pre-commit.yml:33]:
```suggestion
```

**Review by kuhar — APPROVED** [review]
Looks good but let's wait for @ScottTodd to approve

File: .github/workflows/pytest.yml
Comment (kuhar) [inline .github/workflows/pytest.yml:37]: We could install the nightly version:
```
python -m pip install \
  --find-links https://iree.dev/pip-release-links.html \
  --upgrade \
  iree-compiler
```

This may be better if the stable is not updated very often. What do you think, @ScottTodd?

Comment (ScottTodd) [reply .github/workflows/pytest.yml:37]: Stable is currently updated around monthly. I'd prefer to direct downstream projects/users to use the stable releases and use that as a forcing function to get our release process running more regularly/smoothly... but most of our projects are using the nightly releases anyway. Up to you.

File: .github/workflows/pytest.yml
Comment (RattataKing) [reply .github/workflows/pytest.yml]: agree, will not use venv instead

**Review by kuhar — COMMENTED** [review]
Could you paste a link to a successful run on of this workflow on your fork?
Ah no, I can see it actually ran here already but failed.

File: .github/workflows/pytest.yml
Comment (RattataKing) [reply .github/workflows/pytest.yml]: will go with nightly version

File: .github/workflows/pre-commit.yml
Comment (ScottTodd) [inline .github/workflows/pre-commit.yml:8]: I just found out that these should be `tuning/**` (double asterisk). Oops >_>

I also found out that path filters are incompatible with "required checks". More reason to get this code moved into its own repo so the checks can just always run since every PR in the repo will want to run them and ensure they stay passing.

**Review by ScottTodd — COMMENTED** [review]
Workflows LGTM after changing `*` to `**`. Probably want the tests to pass before merging this.

**Review by kuhar — APPROVED** [review]
All tests passes, LGTM

**Pull request merged**
